src/classes/RD2_StatusMappingSettings_CTRL.cls

```diff
@@ -0,0 +1,429 @@
+/*
+    Copyright (c) 2020, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Recurring Donations
+* @description Stores and retrieves Recurring Donation status to status type mapping
+*/
+public with sharing class RD2_StatusMappingSettings_CTRL {
+    /**
+    * @description Custom data mapping exception
+    */
+    private class MappingException extends Exception { }
+
+    /**
+     * @description Interface for the custom metadata type records deployment handler
+     */
+    public interface MetadataDeploymentHandler {
+        String getDeploymentId();
+        Boolean hasDeployResult();
+        Boolean isInProgress();
+        Boolean isSuccess();
+        String getDeployErrorMessage();
+        void save(List<Metadata.CustomMetadata> records);
+        void saveDeploymentId();
+    }
+
+    /**
+     * @description Contains state value to label mapping
+     */
+    private static final Map<String, String> stateLabelByValue = new Map<String, String>{
+        RD2_Constants.STATUS_ACTIVE => System.Label.RD2_StatusMappingStateActive,
```

Comment:
I would prefer that the code uses the actual translated label of the picklist value (these are already packaged). Adding a custom label means that there are two different translations for the same thing, used in two different places. See the `UTIL_Describe.getTranslatedPicklistLabel()` method.